tests/src/services/HeadersTests.scala

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import scala.concurrent.Await
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
+import scala.language.postfixOps
+
+import org.junit.runner.RunWith
+import org.scalatest.Finders
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers
+import org.scalatest.junit.JUnitRunner
+
+import common.WhiskProperties
+import spray.client.pipelining.sendReceive
+import spray.client.pipelining.sendReceive$default$3
+import spray.client.pipelining.unmarshal
+import spray.http.AllOrigins
+import spray.http.BasicHttpCredentials
+import spray.http.HttpHeader
+import spray.http.HttpHeaders.`Access-Control-Allow-Headers`
+import spray.http.HttpHeaders.`Access-Control-Allow-Origin`
+import spray.http.HttpMethod
+import spray.http.HttpMethods.CONNECT
+import spray.http.HttpMethods.DELETE
+import spray.http.HttpMethods.GET
+import spray.http.HttpMethods.HEAD
+import spray.http.HttpMethods.OPTIONS
+import spray.http.HttpMethods.PATCH
+import spray.http.HttpMethods.POST
+import spray.http.HttpMethods.PUT
+import spray.http.HttpMethods.TRACE
+import spray.http.HttpRequest
+import spray.http.HttpResponse
+import spray.http.StatusCodes.Accepted
+import spray.http.StatusCodes.OK
+import spray.http.Uri
+import spray.http.Uri.Path
+import spray.testkit.ScalatestRouteTest
+
+@RunWith(classOf[JUnitRunner])
+class HeadersTests extends FlatSpec
+    with ScalatestRouteTest
+    with Matchers {
+
+    behavior of "Headers at general API"
+
+    val timeout = 60 seconds
+
+    val whiskAuth = WhiskProperties.getBasicAuth
+    val creds = BasicHttpCredentials(whiskAuth.fst, whiskAuth.snd)
+
+    val allMethods = Set(DELETE, GET, POST, PUT)
+    val allowOrigin = `Access-Control-Allow-Origin`(AllOrigins)
+    val allowHeaders = `Access-Control-Allow-Headers`("Authorization", "Content-Type")
+
+    val url = Uri(s"http://${WhiskProperties.getControllerHost}:${WhiskProperties.getControllerPort}")
+    val pipeline: HttpRequest => Future[HttpResponse] = (
+        sendReceive
+        ~> unmarshal[HttpResponse])
+
+    val basePath = Path("/api/v1")
+
+    def toHttpMethod(method: String): Option[HttpMethod] = {
```

Comment:
Can use `HttpMethod.name` to extract the `String` from a `HttpMethod` to do the comparison.